src/importlinter/adapters/user_options.py

```diff
@@ -51,3 +61,38 @@ def _clean_section_config(section_config: Dict[str, Any]) -> Dict[str, Any]:
             else:
                 section_dict[key] = value.strip().split("\n")
         return section_dict
+
+
+class TomlUserOptionReader(ports.UserOptionReader):
+    """
+    Reader that looks for and parses the contents of TOML files.
+    """
+
+    section_name = "importlinter"
+
+    def read_options(self, config_filename: Optional[str] = None) -> Optional[UserOptions]:
+        if not _HAS_TOML:
```

Comment:
An alternative approach would be to do this when we configure the settings in `cli.py`, only passing in the `TomlUserOptionReader` if it's installed. That might simplify the extraction of shared behaviour between `IniFileUserOptionReader` and `TomlUserOptionReader`. Not a blocker though.

Reply:
I see the point but I would prefer the current code because

1. If a reader needs a more complex evaluation it's better to keep this code at the class
2. pylint/flake8 would raise an unused import warning. This can be ignored but is code which is not really needed
3. The toml import in the user_options.py file must still be wrapped in a try-except so it's wuite duplicate